### ISSUE TITLE
Windows 7 tabbar always bright theme

### ISSUE THREAD

**Issue by knoxfighter** [black7375/Firefox-UI-Fix#53]
**Is your feature request related to a problem? Please describe.**
On Windows 7 the header bar is always in lightmode style. Even when using the darkmode it is also white. 
Bugzilla issue for that: https://bugzilla.mozilla.org/show_bug.cgi?id=1702480
Possibly an browser selector can be used, so other designs are not affected.

**Describe the solution you'd like**
Have a workaround for that, so we can see i can see it in dark again.

**Describe alternatives you've considered**
Wait until mozilla is fixing it.

![Unbenannt](https://user-images.githubusercontent.com/18680546/120703954-407da480-c4b6-11eb-8227-8f17b9a5f004.png)

**Comment by black7375** [comment]
Are pure protons the same?

First of all, I don't have a device with Win7 installed.
This is a problem that cannot be solved right away.

**Comment by knoxfighter** [comment]
> Are pure protons the same?
yes, pure proton is the same, that is mostly a bug in firefox. (That's why i added the bugzilla link)

**Comment by black7375** [comment]
```
Whiteboard: [proton-tabs-bar] [priority:2c] → [proton-tabs-bar] [priority:2b]
```
I think Mozilla will fix it.

If they give up, we can try later too.

**Comment by sean1138** [comment]
win 7 here.  
 #TabsToolbar{}:
![image](https://user-images.githubusercontent.com/6720184/121004594-663ccf00-c77e-11eb-912a-457aacb8b89e.png)
:root:-moz-lwtheme #TabsToolbar,
  :root[lwt-default-theme-in-dark-mode] #TabsToolbar {}:
![image](https://user-images.githubusercontent.com/6720184/121004674-8076ad00-c77e-11eb-92d2-b5b4e516ae16.png)
what i get without the extra code:
![image](https://user-images.githubusercontent.com/6720184/121004795-9edca880-c77e-11eb-92c9-2ca7b0b7f6f9.png)

**Comment by sean1138** [comment]
i did some more messing around with some other relevant CSS selectors that i was able to find:
![image](https://user-images.githubusercontent.com/6720184/121009447-b2d6d900-c783-11eb-8bcf-e40f00c1327f.png)
not sure what i'd need to select to get the min/max/close buttons in front of a background-color on the #TabsToolbar.  Perhaps it's not possible and that's why the background-image covered them as well so would need a background image like the afterglow theme has with a cutout for the min/max/close buttons on win7?

I've not yet been successful in finding where I can view the code for the browser, feels like i'm throwing darts blindfolded with my userchrome.css file.

`@media (-moz-os-version: windows-win7) {
  #TabsToolbar{
    background-image: none !important;
    background-color: var(--toolbar-bgcolor) !important;
    color:            var(--toolbar-color) !important;
    z-index: 1 !important;
    border: 1px solid red;
  }
  #TabsToolbar #window-controls {
    border: 1px solid green;
  }
  #TabsToolbar .titlebar-buttonbox-container {
    position: relative !important;
    display: block !important;
      right: 0 !important;
    z-index:10 !important;
    border: 1px solid blue;
  }
}`

**Comment by sean1138** [comment]
i found out about the browser toolbox... and then it stopped opening for me after awhile.  I think i made some progress in getting something interesting for the TabsToolbar area on win7:
![image](https://user-images.githubusercontent.com/6720184/121019156-b328a180-c78e-11eb-9194-e45f87c364b0.png)
```
@media (-moz-os-version: windows-win7) {
	#TabsToolbar{
		background-image: none !important;
		/*border: 1px solid red;*/
	}
	#TabsToolbar .titlebar-spacer[type="pre-tabs"]{
		/*gets rid of transparent space at left*/
		display: none;
	}
	#TabsToolbar .titlebar-spacer[type="post-tabs"]{
		/*gets rid of transparent space at right*/
		display: none;
	}
	#TabsToolbar .toolbar-items{
		/*add space to left after getting rid of .titlebar-spacer*/
		padding-left: 30px;
		/*add space to right after getting rid of .titlebar-spacer*/
		padding-right: 25px;
		background-color: var(--toolbar-bgcolor) !important;
		color:            var(--toolbar-color) !important;
	}
	#TabsToolbar .titlebar-buttonbox-container {
		 background: rgb(255,255,255);
		background: linear-gradient(0deg, var(--toolbar-bgcolor) 20px, rgba(255,255,255,0) 20px) !important;
		display: block !important;
		/*border: 1px solid blue;*/
	}
}
```

**Comment by black7375** [comment]
Interesting.
Let's tweak it a bit more and put it in the next release.

**Comment by sean1138** [comment]
I was able to fix the Browser Toolbox not opening anymore by deleting the chrome_debugger_profile folder in my profile folder, so that's some progress.

But there are issues that i may not be able to figure out, haven't been able to so far at least.
![image](https://user-images.githubusercontent.com/6720184/121038917-2daded00-c7a0-11eb-88fd-1d4eb3d6af1a.png)
1 = there is a 1px border here that i can't figure out what element it is part of.  It looks like it should be part of .titlebar-buttonbox-container but border:none; does not remove it.
2 = .titlebar-buttonbox-container extends 1px above the container visually if it just has a background color - not really an issue probably but something that i noticed when trying different things.

also the 20px gradient stops only look kind of correct for compact density, not the others.  Not sure if there's a better unit (px, %, em, ???) to use here.
![image](https://user-images.githubusercontent.com/6720184/121039867-fb50bf80-c7a0-11eb-9879-d530fb7a0841.png)

**Comment by sean1138** [comment]
screen recording of going through the different customize options:
https://files.catbox.moe/e6otbj.mp4
dark mode normal density:
![image](https://user-images.githubusercontent.com/6720184/121056286-9354a600-c7ad-11eb-9c25-f818189ccc13.png)

the border below the min/max/close buttons ended up being a gap... i think.  I covered it up with an :after psuedo element.

the gradient background to make the space below the min/max/close buttons the right color (instead of windows' aero glass transparency) depending on themes has a different percentage stop for each uidensity.  I could not find a way to change the colors of the min/max/close buttons to match the theme color (var(--toolbar-bgcolor)) instead of windows' aero glass transparency, suspect doing that might be beyond what userchrome.css can do.
```
@media (-moz-os-version: windows-win7) {
	#TabsToolbar{
		/*setting backgrounds here will cover up/hide the min/max/close buttons on win7*/
		background-image: none !important;
		/*background-color: var(--toolbar-bgcolor) !important; */
	}
	#TabsToolbar .titlebar-spacer[type="pre-tabs"]{
		/*gets rid of transparent space at left*/
		display: none;
	}
	#TabsToolbar .titlebar-spacer[type="post-tabs"]{
		/*gets rid of transparent space at right*/
		display: none;
	}
	#TabsToolbar .toolbar-items{
		/*bring back space to left after getting rid of .titlebar-spacer*/
		padding-left: 30px;
		/*bring back space to right after getting rid of .titlebar-spacer*/
		padding-right: 25px;
		background-color: var(--toolbar-bgcolor) !important;
		color:            var(--toolbar-color) !important;
    /*border-bottom:1px solid red; */
	}
	#TabsToolbar .titlebar-buttonbox-container {
		position:relative;
    /*min-width:85px; */
		background: linear-gradient(0deg, var(--toolbar-bgcolor) 60%, rgba(255,255,255,0) 60%);
	}
	#TabsToolbar .titlebar-buttonbox-container::after{
  /*cover up a border/gap appearing at bottom of element*/
		content:'';
		position:absolute;
		top:0;
		left:0;
		right:0;
		bottom:-1px;
		border-bottom:1px solid var(--toolbar-bgcolor);
	}
	[uidensity=compact] #TabsToolbar .titlebar-buttonbox-container {
		background: linear-gradient(0deg, var(--toolbar-bgcolor) 55%, rgba(255,255,255,0) 55%);
	}
	[uidensity=touch] #TabsToolbar .titlebar-buttonbox-container {
		background: linear-gradient(0deg, var(--toolbar-bgcolor) 70%, rgba(255,255,255,0) 70%);
	}
}
```
not sure how many windows 7 users are still out there,  I keep putting off upgrading to win10 because it will take me days to get everything setup the way i want and various programs installed and i don't look forward to that kind of downtime with little to no real benefit.

**Comment by black7375** [comment]
> I was able to fix the Browser Toolbox not opening anymore by deleting the chrome_debugger_profile folder in my profile folder, so that's some progress.

I had the same thing happen to me on Windows.
In that case, refresh.
![image](https://user-images.githubusercontent.com/25581533/121104374-74780300-c7f1-11eb-9d7f-b97d75b32afe.png)

When I have time, I'll also write a guide for contributors.

---

1. This code works?
```css
.titlebar-buttonbox-container .titlebar-buttonbox {
  border-bottom: none !important;
}
```

2. I don't know either.

**Comment by black7375** [comment]
> what i get without the extra code:
> ![default code](https://user-images.githubusercontent.com/6720184/121004795-9edca880-c77e-11eb-92c9-2ca7b0b7f6f9.png)

My personal goal is:
- System Default: Aero glass tab background + white toolbar color
- Light: Tab background should be white. (Color: `var(--lwt-accent-color)`)
- Dark: Tab backgournd should be dark. (Color: `var(--lwt-accent-color)`)
- Alpenglow: Selected tab should have the same transparency as toolbar.

Yesterday I didn't have time to give feedback.

**Comment by black7375** [comment]
While testing, I discovered something I did wrong.

`alpenglow` is my bug.
![image](https://user-images.githubusercontent.com/25581533/121108568-8362b380-c7f9-11eb-8f42-cb1f7f3654aa.png)

**Comment by black7375** [comment]
Based on your results, this value looks good. Maybe..
```css
@media (-moz-os-version: windows-win7) {
  :root:-moz-lwtheme #TabsToolbar,
  :root[lwt-default-theme-in-dark-mode] #TabsToolbar {
    background-image: none !important;
  }

  :root:-moz-lwtheme #TabsToolbar .toolbar-items,
  :root[lwt-default-theme-in-dark-mode] #TabsToolbar .toolbar-items {
    background-color: var(--lwt-accent-color) !important;
    color:            var(--lwt-text-color) !important;
  }

  :root:-moz-lwtheme #TabsToolbar .titlebar-buttonbox-container,
  :root[lwt-default-theme-in-dark-mode] #TabsToolbar .titlebar-buttonbox-container {
    position:relative;
    background: linear-gradient(0deg, var(--lwt-accent-color) 60%, rgba(255,255,255,0) 60%);
  }
  :root[uidensity=compact]:-moz-lwtheme #TabsToolbar .titlebar-buttonbox-container,
  :root[uidensity=compact][lwt-default-theme-in-dark-mode] #TabsToolbar .titlebar-buttonbox-container {
    background: linear-gradient(0deg, var(--lwt-accent-color) 55%, rgba(255,255,255,0) 55%);
  }
  :root[uidensity=touch]:-moz-lwtheme #TabsToolbar .titlebar-buttonbox-container,
  :root[uidensity=touch][lwt-default-theme-in-dark-mode] #TabsToolbar .titlebar-buttonbox-container {
    background: linear-gradient(0deg, var(--lwt-accent-color) 70%, rgba(255,255,255,0) 70%);
  }

  :root:-moz-lwtheme #TabsToolbar .titlebar-buttonbox-container::after,
  :root[lwt-default-theme-in-dark-mode] #TabsToolbar .titlebar-buttonbox-container::after {
    content:'';
    position:absolute;
    top:0;
    left:0;
    right:0;
    bottom:-1px;
    border-bottom:1px solid var(--lwt-accent-color);
  }
}
```

And I'm curious about this result. Can you test it?
```css
@media (-moz-os-version: windows-win7) {
  #navigator-toolbox:-moz-lwtheme {
    background-color: red !important;
  }
}
```

**Comment by sean1138** [comment]
> Btw, what's these value at windows7?
> 
>     * `#TabsToolbar`'s `background-image`
> 
>     * `--lwt-additional-images`
> 
> 
> It doesn't exist on my platform(linux, windows10).

```
@media (-moz-proton) {
/* Add a fog for background tabs to be visible on Windows 7 glass window background */
@media (-moz-os-version: windows-win7) {
  @media (-moz-windows-default-theme) {
    #TabsToolbar:not(:-moz-lwtheme) {
      background-image: radial-gradient(ellipse at bottom, rgba(255,255,255,0.8), rgba(255,255,255,0.5) 80%, transparent);
    }
  }
}
} /*** END proton ***/
```

``` --lwt-additional-images: none;```
both code examples from the browser.css file here.
![image](https://user-images.githubusercontent.com/6720184/121175910-27287f80-c84b-11eb-9a4c-9a861656185f.png)

**Comment by sean1138** [comment]
it would still be my preference to not cover the min/max/close buttons with a background in any theme.  no background on #TabsToolbar, putting the #tabsToolbar background on the .toolbar-items instead achieves this.  should be able to see in the top part of screenshot the browser window's red X/close button is getting covered by the background-image gradient rule.

![image](https://user-images.githubusercontent.com/6720184/121184657-d4ec5c00-c854-11eb-859a-ff324c2f01ca.png)

**Comment by black7375** [comment]
Hello, windows 7 users.

I recently read the CSS source and made it based on it.
Hope it works.
```css
  /** Windows 7 - Compatibility ***********************************************/
  @media (-moz-os-version: windows-win7) {
    /* Header Color */
    :root:not([lwtheme-image]) #navigator-toolbox:-moz-lwtheme {
      background-color: var(--lwt-accent-color) !important;
    }
    #TabsToolbar:-moz-lwtheme {
      color: var(--lwt-text-color) !important;

      /* Remove Aero */
      /* radial-gradient(eclipse at bottom, rgba(255, 255, 255, 0.8), rgba(255, 255, 255, 0.5) 80%, transparent) */
      background-image: unset !important;
    }
  }
```

**Comment by sean1138** [comment]
> Hello, windows 7 users.

![image](https://user-images.githubusercontent.com/6720184/121816900-54f73500-cc4c-11eb-9457-6c4c2b78d3a3.png)

**Comment by black7375** [comment]
Are you using any other effects together?
The code above does not change the color of the maximize, minimize button

**Comment by sean1138** [comment]
> Are you using any other effects together?

no.  i made a new profile with no custom CSS except for what i've copied from https://github.com/black7375/Firefox-UI-Fix/issues/53#issuecomment-860176745 and recorded my screen here:
https://www.youtube.com/watch?v=NE-P424PVW8
> The code above does not change the color of the maximize, minimize button

the code hides/covers up the min/max/close window buttons.  You can see that here along with some explanation of what i see happening:
https://www.youtube.com/watch?v=HksI4SQnIFo
i do not have a good solution other than https://github.com/black7375/Firefox-UI-Fix/issues/53#issuecomment-856092255 where i applied backgrounds to the `#TabsToolbar .toolbar-items` element and not the `#TabsToolbar` element nor the `#navigator-toolbox` element.  you could also apply the same background to the `.titlebar-spacer` elements -- or remove them via `display:none;` and add padding to the `.toolbar-items` to bring back the padding/spacing like i did in that comment/reply.

i stumbled onto searchfox.org the other day.  it lets you search firefox code so it might be helpful if you don't have any of the win7 code on your system, here's a link to search results for "aero" which is the windows 7 transparent glass effect applied to windows that does not exist in win8 or win10 as far as i know.
https://searchfox.org/mozilla-central/search?q=aero&path=
i found comments in `browser/themes/windows/browser.css` referring to `browser/themes/windows/browser-aero.css` but have **not** been able to make anything work the way i want - which is to not have background cover up the min/max/close window buttons in themes that are not aspenglow in a way that doesn't seem hacky or might look broken on someone else's windows 7 because i customized the size of my min/max/close buttons and other window elements.

**Comment by black7375** [comment]
I'll install a VM over the weekend and support it.

**Comment by black7375** [comment]
I've installed the latest commit, and it seems to have reached its goal. https://github.com/black7375/Firefox-UI-Fix/issues/53#issuecomment-856365333
![image](https://user-images.githubusercontent.com/25581533/122665875-7d75b680-d199-11eb-868c-4be2aba10a05.png)
![image](https://user-images.githubusercontent.com/25581533/122665893-a39b5680-d199-11eb-8a1d-7d70ea0f5112.png)
![image](https://user-images.githubusercontent.com/25581533/122665921-c3cb1580-d199-11eb-97c4-b700ed387c38.png)
![image](https://user-images.githubusercontent.com/25581533/122665941-de9d8a00-d199-11eb-8f2b-e2bec5d3cb69.png)
![image](https://user-images.githubusercontent.com/25581533/122665949-f7a63b00-d199-11eb-9479-a834f5479ea7.png)
